scripts/manage-db.py

```diff
@@ -13,22 +14,74 @@
 
 from auslib.db import AUSDatabase
 
+def cleanup_releases(trans, nightly_age, dryrun=True):
+    # This and the subsequent queries use "%%%%%" because we end up going
+    # through two levels of Python string formatting. The first is here,
+    # and the second happens at a low level of SQLAlchemy when the transaction
+    # is being executed.
+    query = """
+LEFT OUTER JOIN releases_history USING (name)
+LEFT JOIN rules ON (name=mapping)
+WHERE name LIKE '%%%%nightly%%%%'
+AND (rules.whitelist <> releases.name OR rules.whitelist IS NULL)
+AND (rules.mapping <> releases.name OR rules.mapping IS NULL)
+AND (timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL %s MONTH) OR change_id is NULL);
+""" % nightly_age
+    if dryrun:
+        todelete = trans.execute("SELECT name FROM releases" + query).fetchall()
+        if todelete:
+            print "Releases rows to be deleted:"
+            print " ".join(itertools.chain(*todelete))
+    else:
+        trans.execute("DELETE releases FROM releases" + query)
+
+
+def cleanup_releases_history(trans, dryrun=True):
+    query = """
+WHERE name LIKE '%%%%latest%%%%'
+AND timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL 14 DAY);
+"""
+    if dryrun:
+        todelete = trans.execute("SELECT name, change_id FROM releases_history" + query).fetchall()
+        if todelete:
+            print "Releases history rows to be deleted:"
+            for key, group in itertools.groupby(todelete, lambda x: x[0]):
+                print "%s: %s" % (key, [int(x[1]) for x in group])
+    else:
+        trans.execute("DELETE releases_history FROM releases_history")
+
+    query = """
+WHERE name NOT LIKE '%%%%latest%%%%' AND name LIKE '%%%%nightly%%%%'
+AND timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL 7 DAY);
+"""
+    if dryrun:
+        todelete = trans.execute("SELECT name, change_id FROM releases_history" + query).fetchall()
+        if todelete:
+            print "Releases history rows to be deleted:"
+            for key, group in itertools.groupby(todelete, lambda x: x[0]):
+                print "%s: %s" % (key, [int(x[1]) for x in group])
+    else:
+        trans.execute("DELETE releases_history FROM releases_history")
+
+
 if __name__ == "__main__":
     from optparse import OptionParser
-    usage = """%s --db dburi action\n""" % sys.argv[0]
+    usage = """%s --db dburi action [options]\n""" % sys.argv[0]
     usage += "Possible actions:\n"
     usage += "  create: Create all the tables required for a new Balrog database\n"
     usage += "  upgrade: Upgrade an existing balrog table to a newer version.\n"
-    usage += "  downgrade: Downgrade an existing balrog table to an older version."
+    usage += "  downgrade: Downgrade an existing balrog table to an older version.\n"
+    usage += "  cleanup: Cleanup old data from a database. Requires an extra arg of maximum age (in months) of releases. Anything older than this will be deleted."
```

Comment:
Nit, change 'of releases' to something that indicates it only affects releases matching '*nightly*'. Bonus points for mentioning the history cleanup.